coderd/database/dbgen/dbgen.go

```diff
@@ -226,6 +226,35 @@ func WorkspaceAgent(t testing.TB, db database.Store, orig database.WorkspaceAgen
 		})
 		require.NoError(t, err, "update workspace agent first connected at")
 	}
+
+	// Add a test antagonist. For every agent we add a deleted sub agent
+	// to discover cases where deletion should be handled.
+	subAgt, err := db.InsertWorkspaceAgent(genCtx, database.InsertWorkspaceAgentParams{
```

Comment:
Review: This may be a bit unusual, but it already helped me catch an issue with the `workspace_prebuilds` VIEW.

Ideally this potential cause would somehow be propagated to the user in case of test failure, but not sure how that could be done. I'll add a log entry to help with discovery.